pkg/services/grafana-apiserver/utils/context.go

```diff
@@ -0,0 +1,83 @@
+package utils
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+
+	"google.golang.org/grpc/metadata"
+	"k8s.io/apiserver/pkg/endpoints/request"
+
+	"github.com/grafana/grafana/pkg/infra/appcontext"
+	grafanaUser "github.com/grafana/grafana/pkg/services/user"
+	"github.com/grafana/grafana/pkg/util"
+)
+
+func ContextWithGrafanaUser(ctx context.Context) (context.Context, error) {
+	// TODO: this relies on grafana populating the context with the user
+	user, err := appcontext.User(ctx)
+	if err != nil {
+		// no user from grafana in context, look for a k8s user
+		info, ok := request.UserFrom(ctx)
+		if !ok {
+			return ctx, fmt.Errorf("could not find k8s user info in context")
+		}
+
+		user = &grafanaUser.SignedInUser{
+			UserID: -1,
+			OrgID:  -1,
+			Name:   info.GetName(),
+			Login:  info.GetName(),
+		}
+
+		if info.GetName() == "system:apiserver" {
+			user.UserID = 1
+			user.OrgID = 1
+		}
+
+		v, ok := info.GetExtra()["user-id"]
```

Comment:
is the user-id/org-id stuff still relevant?  IIUC, we will only get here through the k8s raw path and the user will have `user.SystemPrivilegedGroup` -- should that map to a grafana super admin and everythign else is an error?

(we should also support reading a key from the user -- but that can be set in the handler)